<h1><center>HYPOTHESIS TESTING Assignment-4</center></h1>

## Objective:
To investigate the restaurant owners' claim about the increase in weekly operating costs using hypothesis testing.

## Data Provided:

1. The theoretical weekly operating cost model: W = $(1,000 + 5X)
2. Sample of 25 restaurants with a mean weekly cost of Rs. 3,050
3. Number of units produced in a week (X) follows a normal distribution with a mean (μ) of 600 units and a standard deviation (σ) of 25 units


## Tasks:

### 1. State Hypotheses

The hypotheses define the null and alternative scenarios:
1. Null Hypothesis (H0): The mean weekly operating cost equals the theoretical cost predicted by the model ($ \mu_w = 1000 + 5 \cdot 600 $).
2. Alternative Hypothesis (H1): The mean weekly operating cost is greater than the theoretical cost ($ \mu_w > 1000 + 5 \cdot 600 $).

Given $ X $ has a mean ($ \mu_x $) of 600 units, the theoretical mean weekly cost is:
$$\mu_w = 1000 + 5 \cdot 600 = 1000 + 3000 = 4000 \, \text{(Rs. 4000)}$$

### 2. Calculate the Test Statistic: 

In [3]:
import scipy.stats as stats
import numpy as np

In [5]:
sample_mean = 3050  # Sample mean weekly cost (Rs. 3,050)
mu_x = 600  # Mean number of units produced
sigma_x = 25  # Standard deviation of units produced
n = 25  # Sample size
alpha = 0.05  # Significance level


In [6]:
# Calculate theoretical mean weekly cost: W = 1000 + 5X
# For X = 600, mu_w = 1000 + 5 * 600
mu_w = 1000 + 5 * mu_x

# Standard deviation of weekly cost: sigma_w = 5 * sigma_x
sigma_w = 5 * sigma_x

In [7]:
# Calculate the Test Statistic
# Z = (sample_mean - mu_w) / (sigma_w / sqrt(n))
standard_error = sigma_w / np.sqrt(n)
z_stat = (sample_mean - mu_w) / standard_error

print(f"Theoretical Mean Weekly Cost (μ_w): Rs. {mu_w:.2f}")
print(f"Standard Deviation of Weekly Cost (σ_w): Rs. {sigma_w:.2f}")
print(f"Standard Error: Rs. {standard_error:.2f}")
print(f"Z Test Statistic: {z_stat:.4f}")

Theoretical Mean Weekly Cost (μ_w): Rs. 4000.00
Standard Deviation of Weekly Cost (σ_w): Rs. 125.00
Standard Error: Rs. 25.00
Z Test Statistic: -38.0000


The test statistic is a Z-score, calculated as:
$$Z = \frac{\bar{x} - \mu_w}{\sigma_w / \sqrt{n}}$$
Where:

$ \bar{x} = 3050 $ (Sample mean weekly cost Rs. 3,050).

$ \mu_w = 4000 $ (Theoretical mean weekly cost).

$ \sigma_w = 5 \cdot \sigma_x $  

Standard deviation of weekly cost, since $ W = 1000 + 5X $, and $ \sigma_x = 25 $ (standard deviation of units produced). Thus:
$$\sigma_w = 5 \cdot 25 = 125$$

$ n = 25 $ (Sample size).

Standard error = $ \sigma_w / \sqrt{n} = 125 / \sqrt{25} = 125 / 5 = 25 $.

Now compute the Z-statistic:
$$Z = \frac{3050 - 4000}{25} = \frac{-950}{25} = -38$$

### 3. Determine Critical Value

In [8]:
# 3. Determine the Critical Value
# One-tailed test at alpha = 0.05
critical_value = stats.norm.ppf(1 - alpha)

print(f"Significance Level (α): {alpha}")
print(f"Critical Value (Z_α for one-tailed test): {critical_value:.4f}")

Significance Level (α): 0.05
Critical Value (Z_α for one-tailed test): 1.6449


### 4. Conclusion:

We can observe that the Test Statistic(Z-Statistic) value is less than the Critical Value, we fail to reject the Null Hypotheses. Therefore we fail to reject Null Hypotheses, there is no evidence to support the claim that weekly operating costs are higher than the model predicts. In fact, the sample mean (Rs. 3,050) is lower than the theoretical mean (Rs. 4,000), which contradicts the restaurant owners' claim.